src/databricks/labs/ucx/assessment/crawlers.py

```diff
@@ -61,6 +67,14 @@ def _azure_sp_conf_present_check(config: dict) -> bool:
     return False
 
 
+def _get_azure_spn_application_id(config: dict) -> str:
+    matching_key = [key for key in config.keys() if _AZURE_SP_CONF[2] in key]
```

Comment:
what does `_AZURE_SP_CONF[2]` is pointing to? what if the order of the list items in `_AZURE_SP_CONF` changes?